workspace/pkgs-graph/src/index.ts

```diff
@@ -35,7 +35,14 @@ export function createPkgGraph<T> (pkgs: Array<Package & T>, opts?: {
     unmatched: Array<{ pkgName: string, range: string }>
   } {
   const pkgMap = createPkgMap(pkgs)
-  const pkgMapValues = Object.values(pkgMap)
+  const pkgMapByManifestName: Record<string, Package[] | undefined> = {}
+  for (const key in pkgMap) {
+    const pkg = pkgMap[key]
+    if (pkg.manifest.name) {
+      (pkgMapByManifestName[pkg.manifest.name] ??= []).push(pkg)
```

Comment:
nit: Thoughts on separating the value initialization to a separate line? TypeScript should still be able to infer the later lookup is non-null even if `noUncheckedIndexedAccess` is enabled.

```ts
pkgMapByManifestName[pkg.manifest.name] ??= []
pkgMapByManifestName[pkg.manifest.name].push(pkg
```

I find that slightly more readable, but this is definitely a stylistic preference. Please feel free to disagree and ignore this comment.

Reply:
The name is dynamic so I don't think TS is actually going to figure that out, but I'll try it tomorrow.

Reply:
I've used this pattern in the past — back then I was also surprised to see that TypeScript is able to detect the dynamic object entry initialization.

Double checking in a playground:

https://www.typescriptlang.org/play?noUncheckedIndexedAccess=true#code/MYewdgzgLgBAJgS2FAXDASgU1AJzgHmhwTAHMAaGIk0gbQF0A+GAXhgG8BfAKG4Ho+8JFFoAiAGYgQo+jAD8ctg26JkYydPoA6AA4BXCAAsAFKIBGAQxyiAlEA

Reply:
Also _technically_ speaking, it's also better to initialize this array like:

```js
let array;
for (const foo of bar) {
    if (array === undefined) {
        array = [foo];
    } else {
        array.push(foo);
    }
}
```

Thanks to the way v8 handles arrays, but, I feel like that's overkill.

Reply:
So that pattern (separating out `??=`) doesn't work, actually; the key isn't `const` and so TS won't do the narrowing.